crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,556 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// Grouping of core control parameter of the layout for this node.
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect(PartialEq, Serialize, Deserialize)]
+pub struct LayoutControl {
+    /// Defines how the node will parttake in the layouting.
+    pub display: Display,
+    /// The strategy used to position this node.
+    pub position: Position,
+    /// The inset of this UI node, relative to its default position
+    pub inset: Inset,
+    /// The behavior in case the node overflows its allocated space
+    pub overflow: Overflow,
+}
+
+impl LayoutControl {
+    pub const DEFAULT: Self = Self {
+        display: Display::DEFAULT,
+        position: Position::DEFAULT,
+        inset: Inset::DEFAULT,
+        overflow: Overflow::DEFAULT,
+    };
+}
+impl Default for LayoutControl {
+    fn default() -> Self {
+        Self::DEFAULT
+    }
+}
+
+/// Defines how the node will parttake in the layouting.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect(PartialEq, Serialize, Deserialize)]
+pub enum Display {
+    /// Use Flexbox layout model to determine the position of this [`Node`](crate::ui_node::Node).
+    Flex,
+    /// The node is completely removed from layouting.
+    ///
+    /// The final layout will be calculated as if this node never existed.
+    /// If you want to hide a node and its children, but keep its layout in place,
+    /// set its [`Visibility`](bevy_render::view::Visibility) component instead.
+    None,
+}
+impl Display {
+    pub const DEFAULT: Self = Self::Flex;
+}
+impl Default for Display {
+    fn default() -> Self {
+        Self::DEFAULT
+    }
+}
+
+/// The strategy used to position this node.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Position {
+    /// Positioned in relation to its parent, while considering all sibling-nodes that also have the [`Position::Relative`] value.
+    Relative,
+    /// Positioned in relation to its parent, without considering any sibling nodes.
+    Absolute,
+}
+impl Position {
+    pub const DEFAULT: Self = Self::Relative;
+}
+
+impl Default for Position {
+    fn default() -> Self {
+        Self::DEFAULT
+    }
+}
+
+/// The inset of a UI node from its base position.
+///
+/// Layouting is performed by the Flexbox layout algorithm where the value of [`Inset`] is considered.
```

Comment:
How is it considered?
IMO this needs a bit more explanation what this actually does, especially because this is not a property in CSS (to my knowledge)

Reply:
`Inset` is what we until now has (incorrectly) been calling `position`. It works the same way, it just has been renamed.
https://developer.mozilla.org/en-US/docs/Web/CSS/inset

See https://github.com/DioxusLabs/taffy/issues/271

Better docs never hurt though, so will look into that

Reply:
It is a property in CSS. It's the shorthand name for the `top`, `left`, `right`, and `bottom` properties. Taffy previously called this `position` (and what is now `position` was called `position_type`). They have been renamed to `inset` and `position` to match CSS. https://developer.mozilla.org/en-US/docs/Web/CSS/inset

Reply:
I've updated the doc-comment a bit, hopefully to be a bit clearer.

Reply:
Oops, looks like my search was off.
I like that follow CSS terminology!